### ISSUE TITLE
Misplaced pixels in stb_truetype text rendering after updating from v1.24 to v1.25

### ISSUE THREAD

**Issue by dortamiguel** [nothings/stb#1200]
So before updating this is how text was looking for me

![Screenshot 2021-08-27 at 13 31 33](https://user-images.githubusercontent.com/3449006/131128091-6f0f9f62-2a75-4018-933c-62c960251f0e.jpg)

But after the v1.25 update it started to look like this

![Screenshot 2021-08-27 at 13 33 30](https://user-images.githubusercontent.com/3449006/131128147-7f347435-a637-4459-b4fe-7ef51b1382e6.jpg)

It's a little hard to see but the difference is some "noise" like pixels that are out of place, in this zoom it's easier to see difference

![Screenshot 2021-08-27 at 13 37 28](https://user-images.githubusercontent.com/3449006/131128390-7efac66b-295d-4cad-bd2b-2aa46d5aacb3.jpg)

Left is v24 and right v25.

Here is the code I'm using now to render this text into a texture, maybe I need to update something to make this look nice again?

```c++
const std::wstring content = L"Fullscreen";

float xpos = 0.0f;
int advance, lsb;
for (const auto &character : content)
{
  stbtt_GetCodepointHMetrics(&font.rawFont, character, &advance, &lsb);
  xpos += (advance * font.scale);
}
width = (int)xpos + 1;

int fontHeight = font.ascent + (font.descent * -1);
float scaledFontHeight = (float)fontHeight * font.scale;
height = (int)scaledFontHeight;

std::vector<unsigned char> monoBitmapPixels(
  (size_t)(width * height), (unsigned char)0);

xpos = 0.0f;
float x_shift;
int x0, y0, x1, y1;
for (const auto &character : content)
{
  x_shift = xpos - (float)floor(xpos);

  stbtt_GetCodepointHMetrics(&font.rawFont, character, &advance, &lsb);

  stbtt_GetCodepointBitmapBoxSubpixel(
    &font.rawFont,
    character,
    font.scale,
    font.scale,
    x_shift,
    0,
    &x0,
    &y0,
    &x1,
    &y1);

  auto stride = width * (font.baseline + y0) + (int)xpos + x0;
  stbtt_MakeCodepointBitmapSubpixel(
    &font.rawFont,
    &monoBitmapPixels.at(0) + stride,
    x1 - x0,
    y1 - y0,
    width,
    font.scale,
    font.scale,
    x_shift,
    0,
    character);

  xpos += (advance * font.scale);
}

pixels = std::vector<unsigned char>(
  (size_t)(monoBitmapPixels.size() * RGBA_CHANNEL_AMOUNT), (unsigned char)0);

for (size_t i = 0; i < monoBitmapPixels.size(); i++)
{
  size_t stride = i * 4;
  pixels[stride] = color.r;
  pixels[stride + 1] = color.g;
  pixels[stride + 2] = color.b;
  pixels[stride + 3] = monoBitmapPixels[i];
}
```

**Comment by nothings** [comment]
fixed in 1.26, available now

**Comment by ellipticaldoor** [comment]
thank you, it works fine now
